tests/Configuration/MinimalVersionCheckerTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\Tests\Configuration;
+
+use Rector\Core\Configuration\MinimalVersionChecker;
+use Rector\Core\Exception\Application\PhpVersionException;
+use Symplify\PackageBuilder\Tests\AbstractKernelTestCase;
+
+final class MinimalVersionCheckerTest extends AbstractKernelTestCase
+{
+    /**
+     * @doesNotPerformAssertions
+     */
+    public function testOk(): void
+    {
+        $composerJsonReader = new MinimalVersionChecker\ComposerJsonReader(
+            __DIR__ . '/MinimalVersionChecker/composer-7.2.0.json'
+        );
+        $minimalVersionChecker = new MinimalVersionChecker(
+            '7.5.0',
+            new MinimalVersionChecker\ComposerJsonParser($composerJsonReader->read())
+        );
+
+        $minimalVersionChecker->check();
+    }
+
+    /**
+     * @doesNotPerformAssertions
+     */
+    public function testOkWithSuffix(): void
+    {
+        $composerJsonReader = new MinimalVersionChecker\ComposerJsonReader(
+            __DIR__ . '/MinimalVersionChecker/composer-7.2.0.json'
+        );
+        $minimalVersionChecker = new MinimalVersionChecker(
+            '7.5.0-13ubuntu3.2',
+            new MinimalVersionChecker\ComposerJsonParser($composerJsonReader->read())
+        );
+
+        $minimalVersionChecker->check();
+    }
+
+    public function testExceptionIsThrown(): void
+    {
+        $composerJsonReader = new MinimalVersionChecker\ComposerJsonReader(
+            __DIR__ . '/MinimalVersionChecker/composer-7.2.0.json'
+        );
+        $minimalVersionChecker = new MinimalVersionChecker(
+            '7.1.0',
+            new MinimalVersionChecker\ComposerJsonParser($composerJsonReader->read())
+        );
+
+        $this->expectException(PhpVersionException::class);
+        $minimalVersionChecker->check();
+    }
+
+    public function testExceptionIsThrownWithSuffix(): void
+    {
+        $composerJsonReader = new MinimalVersionChecker\ComposerJsonReader(
+            __DIR__ . '/MinimalVersionChecker/composer-7.2.0.json'
+        );
+        $minimalVersionChecker = new MinimalVersionChecker(
+            '7.1.0-13ubuntu3.2',
+            new MinimalVersionChecker\ComposerJsonParser($composerJsonReader->read())
+        );
+
+        $this->expectException(PhpVersionException::class);
+        $minimalVersionChecker->check();
+    }
```

Comment:
This can be also merged into 1 test() method, with data provider and false/true parameter as for exception or not:

```php
if ($shouldThrowException) {
    $this->expectException(PhpVersionException::class);  
} else {
    $this->doesNotPerformAssertions();
}
```

It would be easier to read & extend later

Reply:
Ready